src/conn.rs

```diff
@@ -527,6 +591,7 @@ impl Conn {
             partition_map: current_partition_map,
             schema: (*current_schema).clone(),
             cache: self.attribute_cache.write().unwrap(),
+            use_caching: true,
```

Comment:
It feels slightly clumsy to `begin_read`, get an `InProgress` with caching on by default and then remember to switch it off. It feels more ergonomic to enable caching as an arg to `begin_read` and pass it through to the creation of the `InProgress`.